src/Apropos/HasParameterisedGenerator.hs

```diff
@@ -76,4 +77,7 @@ genSatisfying :: HasParameterisedGenerator p m => Formula p -> Gen m
 genSatisfying f = do
   label $ fromString $ show f
   s <- element (enumerateScenariosWhere f)
-  parameterisedGenerator s
+  traversalAsGen $ parameterisedGenerator s --TODO this doesn't do shrink containment...
+  -- we can lift a Traversal into Gen
+  -- like GenWrap but for Traversal
+  -- or something...
```

Comment:
This is something we need to thing about. How do we embed a `Traversal` in a `Gen` - we need to do this for `genSatisfying` otherwise we might end up with bad behaviour. I'm undecided as to how to do this though and I think it would be best to leave for another PR.